### PR TITLE
Add julia-nightly.

### PR DESCRIPTION
**Pull request by MurrayT**
Add julia nightly current version.

### CODE REVIEW

File: Casks/julia-nightly.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cask :v1 => 'julia-nightly' do
+  version '0.4.0-90f7cd0980'
+  sha256 'f0fe1adf14f065772473a3bafa35b59a808b142861134396c47f43e4da8db49b'
+
+  # amazonaws.com is the official download host per vendor homepage
+  url "https://s3.amazonaws.com/julianightlies/bin/osx/x64/#{/(\d+\.\d+).*/.match(version)[1]}/julia-#{version}-osx.dmg"
```
Comment (vitorgalvao) [inline Casks/julia-nightly.rb]: We prefer this to me of the form `#{version…}`, so something like `"#{version.sub(%r{(\d+\.\d+).*},'\1')}"`. Same for the others.

File: Casks/julia-nightly.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  name 'Julia'
+  homepage 'http://julialang.org'
+  license :mit
+
+  app "Julia-#{/(.+)-(.+)/.match(version){|m| m[1]+'-dev-'+m[2]}}.app"
+  binary "Julia-#{/(.+)-(.+)/.match(version){|m| m[1]+'-dev-'+m[2]}}.app/Contents/Resources/julia/bin/julia"
+
+  depends_on :macos => '>= 10.7'
```
Comment (vitorgalvao) [inline Casks/julia-nightly.rb]: We prefer this to use the release name (i.e. `depends_on :macos => '>= :lion'`).

File: Casks/julia-nightly.rb (hunk 1 above)
Comment (MurrayT) [inline Casks/julia-nightly.rb]: Ruby isn't my first language, so I don't really get how the regexes work, thanks.

File: Casks/julia-nightly.rb (hunk 2 above)
Comment (MurrayT) [inline Casks/julia-nightly.rb]: I was basing it off of another but changing this is fine.

Comment (vitorgalvao) [inline Casks/julia-nightly.rb]: Could you point me to the other one? It needs to be corrected, then.

Comment (MurrayT) [inline Casks/julia-nightly.rb]: It's the julia one in the homebrew-cask repo. Not a versions one.

**Comment by vitorgalvao** [comment]
Thank you. Could you please [squash your commits](http://davidwalsh.name/squash-commits-git)?

**Comment by fanquake** [comment]
@MurrayT :+1:

**Pull request merged**